src/utils/NumberUtils.h

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#ifndef INCLUDED_NUMBERUTILS_H
+#define INCLUDED_NUMBERUTILS_H
+
+#if defined(_MSC_VER)
+#define really_inline __forceinline
+#else
+#define really_inline inline __attribute__((always_inline))
+#endif
+
+#ifdef USE_FAST_FLOAT
+#include <fast_float/fast_float.h>
+#endif
+
+#include <cstdlib>
+#include <locale>
+#include <system_error>
+#include <type_traits>
+
+namespace NumberUtils
+{
+
+struct Locale
+{
+#ifdef _WIN32
+    Locale() : local(_create_locale(LC_ALL, "C"))
+    {
+    }
+    ~Locale()
+    {
+        _free_locale(local);
+    }
+    _locale_t local;
+#else
+    Locale() : local(newlocale(LC_ALL_MASK, "C", NULL))
+    {
+    }
+    ~Locale()
+    {
+        freelocale(local);
+    }
+    locale_t local;
+#endif
+};
+
+struct from_chars_result
+{
+    const char *ptr;
+    std::errc ec;
+};
+
+static const Locale loc;
+
+#ifdef USE_FAST_FLOAT
+really_inline from_chars_result from_chars(const char *first, const char *last, float &value) noexcept
+{
+    // By design that's not supported by from_chars() so handle it here.
+    if (first && *first == '+')
+        ++first;
+
+    const auto ret = fast_float::from_chars(first, last, value);
+    return {ret.ptr, ret.ec};
+}
+
+really_inline from_chars_result from_chars(const char *first, const char *last, double &value) noexcept
+{
+    // By design that's not supported by from_chars() so handle it here.
+    if (first && *first == '+')
+        ++first;
+
+    const auto ret = fast_float::from_chars(first, last, value);
+    return {ret.ptr, ret.ec};
+}
+
+#else
+really_inline from_chars_result from_chars(const char *first, const char *last, double &value) noexcept
+{
+    errno = 0;
+    if (!first || !last || first == last)
+    {
+        return {first, std::errc::invalid_argument};
+    }
+
+    double tempval{};
+
+    char *endptr = nullptr;
+
+    // Unfortunately we need consistent behaviour
+    // between strtod and from_chars.
+    const std::string temp(first, last - first);
```

Comment:
That kills the performance so, the code must be changed.

Reply:
As explained in the commit message, changing this bit (and all below this review) will make locale and fast_float versions inconsistent.